west.yml: Update Mbed TLS to 2.26.0

d3zd3z opened the pull request:
Move to the latest release of Mbed TLS.

Signed-off-by: David Brown <david.brown@linaro.org>

d3zd3z committed:
west.yml: Update Mbed TLS to 2.26.0

Move to the latest release of Mbed TLS.

Signed-off-by: David Brown <david.brown@linaro.org>

nashif commented on west.yml:
```diff
@@ -88,7 +90,8 @@ manifest:
       path: modules/lib/gui/lvgl
     - name: mbedtls
-      revision: 53044168b43f06ec5654b906a2cdd260bf477edd
+      remote: d3zd3z
```
you should be pointing to the PR in the mbedtls module , not some private remote:

`revision: pull/23/head`

frkv commented:
Adding mbed TLS v2.26.0 is not compatible with TF-M 1.3 (latest release) or TF-M master

The latest current release used in the TF-M project is mbed TLS v2.25

microbuilder commented:
> Adding mbed TLS v2.26.0 is not compatible with TF-M 1.3 (latest release) or TF-M master
> 
> The latest current release used in the TF-M project is mbed TLS v2.25.0

TF-M builds for the secure environment should continue to pull in it's own copy of mbedtls, which will reference 2.25 as of TF-M 1.3.0: https://github.com/zephyrproject-rtos/trusted-firmware-m/blob/master/trusted-firmware-m/config/config_default.cmake#L164

At present, this module is for use with the NS environment, enabling use of the PSA APIs as a backend in mbedtls. Initial testing seems to indicate that 2.26.0 on the NS side is still compatible with TF-M 1.3.0, but if you have seen any specific issues please bring them up so that we can investigate.

There is a goal to bring the entire mbedtls history into Zephyr's mbedtls module repo, which will allow us to reference past tags/releases as well, but that's still a WIP, and 2.26.0 was selected since it is currently the latest release.

ioannisg commented:
I guess it is possible to force TF-M builds in Zephyr to use the mbedtls 2.26 as well. It is a configuration, really, so out-of-tree projects can re-modify this to use the "DOWNLOAD" option.

ceolin commented:
> Adding mbed TLS v2.26.0 is not compatible with TF-M 1.3 (latest release) or TF-M master
> 
> The current release used in the TF-M project is mbed TLS v2.25.0

Yep, TF-M is not the only reason. The other thing that motivated moving to the latest release is that end of life for the current LTS is this year  and this does not work for next Zephyr LTS. Moving to something newer right now allow us find possible failures before next  Zephyr LTS release.

microbuilder reviewed (approved):
Unable to approve the PR on the mbedtls repo, but tested and seems OK here (so far).

d3zd3z committed:
tls: Change some external symbols from Mbed TLS

In Mbed TLS:
    commit eccd88871767e2fba5f3a079cfdfcb77c376cf20
    Author: Gilles Peskine <Gilles.Peskine@arm.com>
    Date:   Tue Mar 10 12:19:08 2020 +0100

        Rename identifiers containing double-underscore

changes the name of a symbol we use.  As part of upgrading to newer
versions of Mbed TLS, change the name of the symbol we use.

A better fix would be to not use this symbol at all, and perhaps define
our own symbol the same way this internal symbol is defined within the
library.

Signed-off-by: David Brown <david.brown@linaro.org>

ioannisg commented:
> Unable to approve the PR on the mbedtls repo, but tested and seems OK here (so far).

The crypto.mbedtls test seems to be failing, though

microbuilder commented:
@frkv Just as a FYI, I did a test pointing TF-M on the S side to mbedtls 2.26.0 and it still builds and runs as expected.
